lib/plugin-registry.coffee

```diff
@@ -35,22 +36,25 @@ module.exports =
     @plugins[code] = plugin
 
     if @enabled
-      @addConfigForPlugin code, plugin, key
+      @addConfigForPlugin(code, plugin, key)
       @observePlugin code, plugin, key
 
   addConfigForPlugin: (code, plugin, key) ->
+    return if atom.config.get(key) != undefined
     @config.plugins.properties[code] =
       type: 'boolean',
       title: plugin.title,
       description: plugin.description,
       default: true
 
-    if atom.config.get(key) == undefined
-      atom.config.set key, @config.plugins.properties[code].default
+    atom.config.set key, @config.plugins.properties[code].default
 
   observePlugin: (code, plugin, key) ->
     @subscriptions.add atom.config.observe(
       key, (isEnabled) =>
+        if plugin.name? and atom.packages.isPackageDisabled(plugin.name) and isEnabled
+            return atom.config.set(key, false)
+
```

Comment:
I think this should be resolved in other way, it requires the package name to be the same the plugin name. What would happen with packages with multiple plugins?
Also, the fix for this should be apply for canvas effect and flow as well.
Not sure yet how to fix this issue, probably a new set of unregister* functions.